src/box/iproto.cc

```diff
@@ -730,6 +745,13 @@ struct iproto_connection
 	char salt[IPROTO_SALT_SIZE];
 	/** Iproto connection thread */
 	struct iproto_thread *iproto_thread;
+	/** Link in iproto_thread->connections. */
+	struct rlist in_connections;
+	/**
+	 * The connection is processing replication command so that
+	 * IO is handled by relay code.
+	 */
+	bool is_in_replication;
```

Comment:
I don't like that we try to stop replication connections the same way as other iproto connections. I think we should introduce a separate function that would stop relays and call it instead of forcefully cancelling the fibers handling replication connections. Rationale: relays may not have proper cancellation points; e.g. a relay fiber may be uninterruptedly waiting for some other thread, which should be stopped first.

Reply:
Discussed f2f and decided keep this way. See the next comment.